src/bin/laminar-tester.rs

```diff
@@ -123,33 +146,32 @@ fn run_client(test_name: &str, destination: &str, endpoint: &str, pps: &str, tes
 }
 
 // Basic test where the client sends packets at a steady rate to the server
-fn test_steady_stream(sender: &Sender<Packet>, target: &str, pps: &str, test_duration: &str) {
+fn test_steady_stream(sender: &Sender<Packet>, config: ClientConfiguration) {
     info!("Beginning steady-state test");
-    let data_to_send = String::from("steady-state test packet");
-    let server_addr: SocketAddr = target.to_socket_addrs().unwrap().next().unwrap();
-    let pps = pps.parse::<u64>().unwrap();
-    let test_duration = test_duration.parse::<u64>().unwrap();
-    let test_duration = Duration::from_secs(test_duration);
+
     let test_packet = Packet::new(
-        server_addr,
-        data_to_send.into_bytes().into_boxed_slice(),
+        config.listen_host,
+        config.test_name.into_bytes().into_boxed_slice(),
         DeliveryMethod::ReliableUnordered,
     );
-    let time_quantum = 1000 / pps;
+
+    let time_quantum = 1000 / config.packet_ps as u64;
     let start_time = Instant::now();
     let mut packets_sent = 0;
+
     loop {
         sender
             .send(test_packet.clone())
             .expect("Unable to send a client packet");
+
         packets_sent += 1;
-        let now = Instant::now();
-        let d = now - start_time;
-        if d >= test_duration {
+
+        if start_time.elapsed() >= config.maximal_duration {
```

Comment:
This should probably be an error, since we expect it to not exceed the max duration?

Reply:
Not sure what you mean with returning an error, the client should just stop sending when some time has elapsed.

Reply:
Ah, ok, so maximal duration is the amount of time the test should last, not a safety valve.